jetty-server/src/main/java/org/eclipse/jetty/server/Request.java

```diff
@@ -2391,7 +2385,7 @@ else if (oldQueryParams == null || oldQueryParams.size() == 0)
                 setQueryString(oldQuery);
             else if (oldQuery == null)
                 setQueryString(newQuery);
-            else
+            else if (oldQueryParams.keySet().stream().anyMatch(newQueryParams.keySet()::contains))
```

Comment:
Couldn't / shouldn't the code of this branch (as well as the new `else` one) be moved to `URIUtil`?

Reply:
Hmmm  probably.... but now I'm losing confidence that this code is even necessary in the first place!  The actual parameters maps are merged above with:
```java
            mergedQueryParams = new MultiMap<>(newQueryParams);
            mergedQueryParams.addAllValues(oldQueryParams);
```

So surely the actual query string itself should be merged with just
```java
    setQueryString(newQuery + '&' + oldQuery);
```

I think I'll try removing that complex logic and see what fails and then check with the TCK if it should or shouldn't fail like that!

Reply:
Ugh!  @janbartel just pointed me at #2074 so we dropped the ball on this a bit as we do have different behaviours and we've not chosen one or the other.   I'm going to experiment with the simplest interpretation and see what the TCK says

Reply:
Actually cancel that.  I'm going to merge this one as it is mostly unrelated and work on #2074 separately